ios/PurchasesHybridCommon/PurchasesHybridCommonTests/PurchasesHybridCommonTests.swift

```diff
@@ -40,137 +49,136 @@ class PurchasesHybridCommonTests: QuickSpec {
+                let mockCreated = Bool.random()
+                mockPurchases.stubbedLogInCompletionResult = (self.mockCustomerInfo, mockCreated, nil)
 
-//            it("returns error if not successful") {
-//                let mockPurchases = MockPurchases()
-//                let mockError = NSError(domain: "revenuecat", code: 123)
-//
-//                mockPurchases.stubbedLogInCompletionResult = (nil, false, mockError)
-//
-//                Purchases.setDefaultInstance(mockPurchases)
-//                var receivedResultDict: NSDictionary?
-//                var receivedError: ErrorContainer?
-//
-//                let appUserID = "appUserID"
-//                RCCommonFunctionality.logIn(withAppUserID: appUserID) { resultDict, error in
-//                    receivedResultDict = resultDict as NSDictionary?
-//                    receivedError = error
-//                }
-//
-//                let expectedErrorDict: NSDictionary = [
-//                    "code": mockError.code,
-//                    "message": mockError.localizedDescription,
-//                    "underlyingErrorMessage": ""
-//
-//                ]
-//
-//                expect { receivedResultDict }.to(beNil())
-//                expect { receivedError }.toNot(beNil())
-//                expect { receivedError!.error as NSError } == mockError
-//                expect { receivedError!.code } == mockError.code
-//                expect { receivedError!.message } == mockError.localizedDescription
-//                expect { receivedError!.info as NSDictionary } == expectedErrorDict
-//            }
+                Purchases.setDefaultInstance(mockPurchases)
+
+                let appUserID = "appUserID"
+                CommonFunctionality.logIn(appUserID: appUserID) { _, _ in }
+
+                expect { mockPurchases.invokedLogInCount } == 1
+                expect { mockPurchases.invokedLogInParameters?.appUserID } == appUserID
+            }
+
+            it("returns customerInfo and created if successful") {
+                let mockPurchases = MockPurchases()
+                let mockCreated = Bool.random()
+                mockPurchases.stubbedLogInCompletionResult = (self.mockCustomerInfo, mockCreated, nil)
+
+                Purchases.setDefaultInstance(mockPurchases)
+                var receivedResultDict: NSDictionary?
+                var receivedError: ErrorContainer?
+
+                let appUserID = "appUserID"
+                CommonFunctionality.logIn(appUserID: appUserID) { resultDict, error in
+                    receivedResultDict = resultDict! as NSDictionary
+                    receivedError = error
+                }
+
+                // todo: update this once APIs get updated to v4
+                let expectedResult: NSDictionary = [
+                    "purchaserInfo": self.mockCustomerInfo.dictionary,
+                    "created": mockCreated
+                ]
+
+                expect { receivedResultDict } == expectedResult
```

Comment:
Huh I never thought to pass a closure to the `@autoclosure` parameters. Any reason not to use `expect(receivedResultDict)` and rely on the auto closure?

Reply:
Good catch! 
This is just copy/pasta. We use the closure to check for throws, so likely this got copy/pasted from something that was testing that a block throws and we never saw it (this is fairly old code)

Reply:
updated!